pkg/provider/core.go

```diff
@@ -430,6 +441,28 @@ func FindInstanceByFolderAndName(ctx context.Context, sdk *ycsdk.SDK, folderID s
 	return result.Instances[0], nil
 }
 
+func FindUnusedAddressFromListByFolder(ctx context.Context, sdk *ycsdk.SDK, folderID string, addresses []string) (string, error) {
```

Comment:
Minor nit. This function can be unexpected function.
```suggestion
func findUnusedAddressFromListByFolder(ctx context.Context, sdk *ycsdk.SDK, folderID string, addresses []string) (string, error) {
```